### ISSUE TITLE
Rock vapour calculation

### ISSUE THREAD

**Issue by nichollsh** [FormingWorlds/PROTEUS#215]
Currently, PROTEUS only supports volatile outgassing, but at higher temperatures we should expect significant amounts of rock vapour to be produced. Integrating these into the radiative transfer would be complex, but they could quite easily be incorporated into calculations of MMW and bulk density.

We can calculate vapour partial (surface) pressures at a given temperature and fO2 using codes such as Vaporock and Lavatmos. These codes are somewhat annoying to setup, since they depend on ThermoEngine, but I have been successful in doing this on my Mac laptop.

We would have to think carefully about how to include these gases alongside the currently supported volatiles.

**Comment by timlichtenberg** [comment]
Agree, we had similar issues in Björn Koops' BSc thesis earlier this year when trying out Vaporock. At the meeting on Wednesday, we should discuss whether pre-computed look-up tables may provide an entry. 

Why do you think that integrating them into the radiative transfer would be complex? If we have the opacities for the major gases, SOCRATES should be able to compute k-tables?

**Comment by nichollsh** [comment]
Pre-computing this is a nice idea, particularly if we are assuming BSE composition since that would only leave 2 variables (Tsurf, fO2) for the table's axes.

I think it might be complex because SOCRATES doesn't currently support any refractory species. It uses hardcoded gas IDs defined in [gas_list_pcf.f90](https://github.com/nichollsh/SOCRATES/blob/4d7ec2afe8b26671987aa1a678362c6834b8e19b/src/radiance_core/gas_list_pcf.F90#L26), so we would need to manually add all of the required gases to this list. This is easy to do, but it would mean diverging from the Met Office SOCRATES distribution somewhat.

**Comment by nichollsh** [comment]
The currently supported refractory species are: TiO, VO, Na, K, FeH, CrH, Li, Rb, Cs

**Comment by timlichtenberg** [comment]
Mh, yeah, these are clearly not sufficient. But yes, perhaps assuming BSE and condensing the axes would make sense. I wonder if the axes should be (Tsurf, Psurf, fO2), but that's a thing to discuss on Wednesday with Yamila's group further.

**Comment by nichollsh** [comment]
Notes from meeting with Yamila's group:
* Lavatmos can calculate rock vapour partial pressures - can probably use a look up table to do this easily.
* Installing lavatmos might be a pain - depends on ThermoEngine.
* We can probably integrate this into CALLIOPE directly.
* Include Lavatmos in the main CALLIOPE solver loop where we calculate elemental mass inventories from partial pressure guesses. Only possible if we allow the fO2 to evolve at the same time
* fO2 will need to be made into a dependent variable, which is good anyway
* Need to think about which opacities to prioritise
* Would be good to get Lavatmos integrated into a framework like PROTEUS
